cmake/extensions.cmake

```diff
@@ -2245,3 +2279,379 @@ function(target_byproducts)
                      COMMENT "Logical command for additional byproducts on target: ${TB_TARGET}"
   )
 endfunction()
+
+########################################################
+# 4. Zephyr linker function
+########################################################
+# 4.1. zephyr_linker*
+#
+# The following methods are for defining linker structure using CMake functions.
+#
+# This allows Zephyr developers to define linker sections and their content and
+# have this configuration rendered into an appropriate linker script based on
+# the toolchain in use.
+# For example:
+# ld linker scripts with GNU ld
+# ARM scatter files with ARM linker.
+#
+# Example usage:
+# zephyr_linker_section(
+#   NAME my_data
+#   VMA  RAM
+#   LMA  FLASH
+# )
+#
+# and to configure special input sections for the section
+# zephyr_linker_section_configure(
+#   SECTION my_data
+#   INPUT   "my_custom_data"
+#   KEEP
+# )
+
+
+# Usage:
+#   zephyr_linker([FORMAT <format>)
+#
+# Zephyr linker general settings.
+# This function specifies general settings for the linker script to be generated.
+#
+# FORMAT <format>: The output format of the linked executable.
```

Comment:
The function `zephyr_linker()` itself should support being called more than once, but with different arguments.

Currently it only handles output format, but another use-case could be a default alignment.
Today, we do use `_region_min_align` but at current stage, i have not yet decided myself how this is to be handled in the new scheme, but setting this through `zephyr_linker` could be an option, and that setting could be set independently of `FORMAT`.

But calling `zephyr_linker(FORMAT <val>)` twice should be an error, so that will be fixed.